src/rust/engine/fs/src/store.rs

```diff
@@ -343,7 +347,7 @@ impl Store {
             }).collect::<Vec<_>>(),
         ).and_then(future::join_all)
         .map(|uploaded_digests| (uploaded_digests, ingested_digests))
-      }).map(|(uploaded_digests, ingested_digests)| {
+      }).map(move |(uploaded_digests, ingested_digests)| {
```

Comment:
Not sure it super matters, but might be worth stopping the clock (i.e. moving `start_time.elapsed()`) before mapping over the digests. Again, not sure it super matters.

Reply:
I think I'm going to optimistically skip this for now, in the hopes that the event loop turns fast enough that this doesn't matter; if that turns out not to be the case, we have big problems...